### PR TITLE
1552 global search 

### PR DESCRIPTION
**Pull request by kenjinp**
# Add global search with configuration and simple UI

## Known issues
- When typing in the search bar, the first of the autocompleted options is not selected by default, so you have to press enter twice
- The results list grid sizes aren't so pretty

## Still TODO
- remove `query-string` dependency (switched to `qs` for nested object support)
- Table View

note: made a new branch and PR that is branched from `master` instead of `fusion_dev`

resolves: https://github.com/BlueBrain/nexus/issues/1552

### CODE REVIEW

File: src/shared/types/search.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export interface SearchResponse<T> {
+  took: number;
+  timed_out: boolean;
+  _scroll_id?: string;
+  // _shards: ShardsResponse;
```
Comment (Stafeeva) [inline src/shared/types/search.ts:5]: forgotten comment?

File: src/shared/hooks/useSearchQuery.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+    // TODO fix query to match @id's as well
+    const matchQuery = query
+      ? ['wildcard', '_original_source', `${query}*`]
+      : ['match_all', {}];
+
+    const body = bodybuilder();
+
+    body
+      // @ts-ignore
```
Comment (Stafeeva) [inline src/shared/hooks/useSearchQuery.ts:110]: Just checking you didn't forget to remove it :)

File: src/shared/types/search.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    };
+    max_score: number;
+    hits: {
+      _index: string;
+      _type: string;
+      _id: string;
+      _score: number;
+      _source: T;
+      _version?: number;
+      // _explanation?: Explanation;
```
Comment (Stafeeva) [inline src/shared/types/search.ts:18]: same as above

File: src/subapps/search/components/ActiveFilters/index.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import './ActiveFilters.less';
+
+const ActiveFilters: React.FC<{
+  searchProps: UseSearchProps;
+  onClear: () => void;
+  onClearFacet: (key: string, value: string) => void;
+  onClearQuery: () => void;
+}> = ({ searchProps, onClear, onClearQuery, onClearFacet }) => {
+  const tagList = [
+    <span key="clear-filters" style={{ display: 'inline-block' }}>
```
Comment (Stafeeva) [inline src/subapps/search/components/ActiveFilters/index.tsx:16]: styles can be moved to `ActiveFilters.less`

File: src/subapps/search/components/ActiveFilters/index.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        onClick={onClear}
+      >
+        Clear all filters
+      </Button>
+    </span>,
+    ...[
+      !!searchProps.query ? (
+        <span
+          key={`search-${searchProps.query}`}
+          style={{ display: 'inline-block' }}
```
Comment (Stafeeva) [inline src/subapps/search/components/ActiveFilters/index.tsx:29]: styles can be moved to ActiveFilters.less

File: src/subapps/search/components/ActiveFilters/index.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          ([key, { propertyKey, type, label: filterGroupLabel, value }]) => {
+            return Array.from(value).map(val => {
+              const [label] = val.split('/').reverse();
+              const handleClearFacet = () => {
+                onClearFacet(key, val);
+              };
+              return (
+                <span
+                  key={`${propertyKey}-${val}`}
+                  style={{ display: 'inline-block' }}
```
Comment (Stafeeva) [inline src/subapps/search/components/ActiveFilters/index.tsx:53]: same as above please

File: src/subapps/search/views/SearchView.less [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  max-width: 400px;
+  &:hover {
+    .highShadow();
+    transform: scale(1.05);
+  }
+  .description {
+    position: relative;
+    max-height: 250px;
+    overflow: hidden;
+    // border: 1px solid @border-color-base;
```
Comment (Stafeeva) [inline src/subapps/search/views/SearchView.less:23]: does it need to be here? :)

File: src/subapps/search/views/SearchView.tsx [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+          <Row style={{ padding: '0 1em' }}>
+            <ActiveFilters
+              searchProps={searchProps}
+              onClearQuery={handleClearQuery}
+              onClearFacet={handleClearFacet}
+              onClear={handleClearFilters}
+            />
+          </Row>
+          <Row>
+            {/* <Spin
```
Comment (Stafeeva) [inline src/subapps/search/views/SearchView.tsx:297]: Do we plan to use it?

**Review by Stafeeva — DISMISSED** [review]

File: src/shared/hooks/useSearchQuery.ts (hunk 2 above)
Comment (kenjinp) [reply src/shared/hooks/useSearchQuery.ts:111]: we'll have to update ts to support what I'm trying to do here, I wrote a comment

**Review by adulbrich — APPROVED** [review]

**Review by Stafeeva — APPROVED** [review]

**Pull request merged**